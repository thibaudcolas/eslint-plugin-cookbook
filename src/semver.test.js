const pkg = require("../package.json");
const config = require("./recommended.json");

describe("semver - should those tests break, consider releasing a new major version of the package", () => {
  it("dependencies", () => {
    expect(pkg.dependencies).toMatchInlineSnapshot(`
Object {
  "eslint-plugin-import": "^2.14.0",
  "eslint-plugin-jsx-a11y": "^6.1.1",
  "eslint-plugin-react": "^7.11.1",
}
`);
  });

  it("peerDependencies", () => {
    expect(pkg.peerDependencies).toMatchInlineSnapshot(`
Object {
  "eslint": "^5.6.0",
}
`);
  });

  it("devDependencies", () => {
    const devDependencies = Object.keys(pkg.devDependencies)
      .filter(
        (dep) =>
          dep === "eslint" ||
          dep.startsWith("eslint-config") ||
          dep.startsWith("eslint-plugin"),
      )
      .reduce((deps, key) => {
        deps[key] = pkg.devDependencies[key];
        return deps;
      }, {});
    expect(devDependencies).toMatchInlineSnapshot(`
Object {
  "eslint": "^5.6.0",
  "eslint-config-airbnb": "^17.1.0",
  "eslint-config-prettier": "^3.0.1",
}
`);
  });

  it("config", () => {
    expect(config).toMatchSnapshot();
  });
});
