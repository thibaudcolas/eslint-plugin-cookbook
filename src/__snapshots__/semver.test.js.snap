// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`semver - should those tests break, consider releasing a new major version of the package custom config 1`] = `
Object {
  "env": Object {
    "browser": true,
    "commonjs": true,
    "es6": true,
    "jest": true,
    "node": true,
  },
  "extends": Array [
    "airbnb",
  ],
  "rules": Object {
    "arrow-body-style": Array [
      0,
    ],
    "eol-last": Array [
      1,
    ],
    "id-length": Array [
      0,
    ],
    "import/no-cycle": Array [
      2,
    ],
    "import/no-extraneous-dependencies": Array [
      2,
      Object {
        "devDependencies": true,
        "optionalDependencies": false,
        "peerDependencies": false,
      },
    ],
    "import/prefer-default-export": Array [
      0,
    ],
    "indent": Array [
      2,
      4,
    ],
    "jsx-a11y/label-has-for": Array [
      0,
    ],
    "jsx-quotes": Array [
      1,
      "prefer-double",
    ],
    "max-len": Array [
      1,
      120,
      4,
      Object {
        "ignoreUrls": true,
      },
    ],
    "no-console": Array [
      1,
      Object {
        "allow": Array [
          "warn",
          "error",
          "time",
          "timeEnd",
        ],
      },
    ],
    "no-multi-spaces": Array [
      0,
    ],
    "no-new": Array [
      1,
    ],
    "no-param-reassign": Array [
      2,
      Object {
        "props": false,
      },
    ],
    "no-plusplus": Array [
      0,
    ],
    "no-warning-comments": Array [
      1,
      Object {
        "location": "start",
        "terms": Array [
          "todo",
          "fixme",
          "xxx",
        ],
      },
    ],
    "object-shorthand": Array [
      2,
      "methods",
    ],
    "react/forbid-prop-types": Array [
      2,
      Object {
        "forbid": Array [
          "any",
        ],
      },
    ],
    "react/jsx-boolean-value": Array [
      0,
    ],
    "react/jsx-filename-extension": Array [
      2,
      Object {
        "extensions": Array [
          ".js",
        ],
      },
    ],
    "react/jsx-indent": Array [
      2,
      4,
    ],
    "react/jsx-indent-props": Array [
      2,
      4,
    ],
    "react/jsx-no-bind": Array [
      0,
    ],
    "react/sort-comp": Array [
      0,
    ],
  },
}
`;

exports[`semver - should those tests break, consider releasing a new major version of the package dependencies 1`] = `
Object {
  "eslint-plugin-import": "^2.14.0",
  "eslint-plugin-jsx-a11y": "^6.1.1",
  "eslint-plugin-prettier": "^2.6.2",
  "eslint-plugin-react": "^7.11.1",
}
`;

exports[`semver - should those tests break, consider releasing a new major version of the package devDependencies 1`] = `
Object {
  "eslint": "^5.6.0",
  "eslint-config-airbnb": "^17.1.0",
  "eslint-config-prettier": "^3.0.1",
}
`;

exports[`semver - should those tests break, consider releasing a new major version of the package peerDependencies 1`] = `
Object {
  "eslint": "^5.6.0",
}
`;
